File: Backend/api/v1/tools/password.py
from fastapi import APIRouter
from typing import Dict, Any
import secrets
import string

from schemas.request.password import PasswordGenerateRequest
from schemas.response.password import PasswordGenerateResponse
from services.password_service import PasswordService
from core.exceptions import ValidationException

router = APIRouter(prefix="/password", tags=["password"])


@router.post("/generate", summary="生成安全密码", response_model=PasswordGenerateResponse)
async def generate_password(request: PasswordGenerateRequest) -> Dict[str, Any]:
    """
    生成安全密码

    可以指定密码长度和包含的字符类型（大小写字母、数字、符号）
    """
    password = PasswordService.generate_password(
        length=request.length,
        include_symbols=request.include_symbols,
        include_numbers=request.include_numbers,
        include_uppercase=request.include_uppercase,
        include_lowercase=request.include_lowercase
    )

    return {
        "success": True,
        "password": password,
        "length": request.length,
        "character_types": {
            "lowercase": request.include_lowercase,
            "uppercase": request.include_uppercase,
            "numbers": request.include_numbers,
            "symbols": request.include_symbols
        }
    }

File: Backup/tools_project_rust/src/models/password.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct PasswordGenerateRequest {
    #[serde(default = "default_length")]
    pub length: usize,
    #[serde(default = "default_true")]
    pub include_symbols: bool,
    #[serde(default = "default_true")]
    pub include_numbers: bool,
    #[serde(default = "default_true")]
    pub include_uppercase: bool,
    #[serde(default = "default_true")]
    pub include_lowercase: bool,
}

fn default_length() -> usize { 12 }
fn default_true() -> bool { true }

#[derive(Debug, Serialize, Deserialize)]
pub struct CharacterTypes {
    pub lowercase: bool,
    pub uppercase: bool,
    pub numbers: bool,
    pub symbols: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct PasswordGenerateResponse {
    pub password: String,
    pub length: usize,
    pub character_types: CharacterTypes,
}
